pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -751,9 +753,10 @@ func IsServiceInternal(service *corev1.Service) bool {
 	return false
 }
 
-// loadBalancerServiceTagsModified verifies that none of the managedAnnotations have been changed and also the AWS tags annotation
+// loadBalancerServiceTagsModified verifies that none of the managedAnnotations except awsLBAdditionalResourceTags have been changed
 func loadBalancerServiceTagsModified(current, expected *corev1.Service) (bool, *corev1.Service) {
-	ignoredAnnotations := managedLoadBalancerServiceAnnotations.Union(sets.NewString(awsLBAdditionalResourceTags))
+	ignoredAnnotations := managedLoadBalancerServiceAnnotations.Clone()
+	ignoredAnnotations.Delete(awsLBAdditionalResourceTags)
```

Comment:
Thank you for the detailed explanation on how the status logic works and how it sets `Upgradeable=False`, as well as the exception that existed with `awsLBAdditionalResourceTags` before this PR. Earlier, I was under the impression that the status logic would still set `Upgradeable=False` even if `awsLBAdditionalResourceTags` was updated by the controller.


> So now that you are making the awsLBAdditionalResourceTags annotation a managed annotation, don't we still want to set Upgradeable=False if the annotation value doesn't match the expected value?

Since `awsLBAdditionalResourceTags` will now be managed by the controller, and we still want to set `Upgradeable=False` if it’s updated by something other than the ingress controller, it does indeed make sense to use `managedLoadBalancerServiceAnnotations` directly in this logic. This way, the status logic will behave consistently for managed annotations when any discrepancy is observed.

I've removed the `loadBalancerServiceTagsModified()` function and used `loadBalancerServiceAnnotationsChanged()` directly inside `loadBalancerServiceIsUpgradeable()` and also added some comments for clearer understanding of the flow.